gcamwrapper/query_library.py

```diff
@@ -0,0 +1,251 @@
+import yaml
+import warnings
+import re
+import os.path as path
+import pkg_resources
+
+def read_yaml(yaml_file):
+    '''Read a YAML file.
+       
+    :param yaml_file: The file path to the YAML file to load
+    :type yaml_file: str
+
+    :returns: A dict representing the YAML data
+    '''
+
+    with open(yaml_file, 'r') as yml:
+        return yaml.load(yml, Loader=yaml.FullLoader)
+
+'''The parsed query file included with the package'''
+PACKAGE_QUERIES = yaml.load(pkg_resources.resource_stream(__name__, 'query_library.yml'), Loader=yaml.FullLoader)
```

Comment:
We should convert this global to a constrained scope.  In this case, you are only calling this from `get_query` so I believe you can set it up as the following after making a data dir in your `gcamwrapper` package and adding the `query_library.yml` into it (don't forget to add it it the MANIFEST.in file and set `include_package_data` to `True` in your setup.py) in your `get_query` function and dump the global:

```python
    if query_file is None:
        queries = yaml.load(pkg_resources.resource_filename('gcamwrapper', 'data/query_library.yml'), 
    else:
        queries = read_yaml(query_file)
    query = queries
```

Reply:
I think what I was trying to accomplish is a little different than what you have here.  What I was trying to accomplish was to load the internal query YAML file just once and not every time a user calls `get_query`.  In R you were correct that the `PACKAGE_QUERIES` gets loaded into the global namespace -- which I should have known because I have dealt with this before in other packages.

By experimenting with python is doesn't seem to behave in quite the same way:

```{python}
>> import gcamwrapper as gw
>>> PACKAGE_QUERIES
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'PACKAGE_QUERIES' is not defined
>>> gw.PACKAGE_QUERIES
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'gcamwrapper' has no attribute 'PACKAGE_QUERIES'
>>> from gcamwrapper.query_library import PACKAGE_QUERIES
>>> PACKAGE_QUERIES
{'electricity': {'share_weight': ['world/region{region@name}/sector[NamedFilter,StringEquals,electricity]/subsector{subsector@name}/share-weight{year@year}', 'unitless']}, 'socioeconomic': {'labor_productivity': ['world/region{region@name}/GDP/laborproductivity{year@year}', 'unitless'], 'population': ['world/region{region@name}/demographic/population{year@year}/totalPop', 'thous ppl']}, 'emissions': {'co2_emissions': ['world/region{region@name}//ghg[NamedFilter,StringEquals,CO2]/emissions{year@year}', 'MTC']}, 'ag': {'prices': ['world/region{region@name}/sector{sector@name}/price{year@year}', '1975$/kg'], 'yield': ['world/region{region@name}/sector{sector@name}/subsector/technology{tech@name}/period{year@year}/yield', 'Mt/thous km2']}, 'land': {'land_allocation': ['world/region{region@name}//child-nodes{leaf@name}/land-allocation{year@year}', 'thous km2']}, 'test': {'fail': ['world/region{region@name}/GDP/laborproductivity{year@year}/somethingelse{region@name}', 'unitless', 'extra'], 'fail2': ['world/region{region@name}/GDP/laborproductivity{year@year}/somethingelse{some@fail}', 'unitless', 'extra']}}
>>> PACKAGE_QUERIES = dict()
>>> gw.get_query("electricity", "share_weight")
'world/region{region@name}/sector[NamedFilter,StringEquals,electricity]/subsector{subsector@name}/share-weight{year@year}'
>>> PACKAGE_QUERIES
{}
```

This behavior seems to be what I would want.  Is there some pitfall that I am missing?

Reply:
looks good